tests/tracked_fn_multiple_args.rs

```diff
@@ -0,0 +1,25 @@
+//! Test that a `tracked` fn on multiple salsa struct args
+//! compiles and executes successfully.
+
+#[salsa::input]
+struct MyInput {
+    field: u32,
+}
+
+#[salsa::interned]
+struct MyInterned<'db> {
+    field: u32,
+}
+
+#[salsa::tracked]
+fn tracked_fn<'db>(db: &'db dyn salsa::Database, input: MyInput, interned: MyInterned<'db>) -> u32 {
```

Comment:
Why this particular test? Tracked functions with multiple arguments today compile to a tracked function on a hidden interned value -- is that what you were intending to test?

Reply:
Yes. Though this PR is not directly related to that, but as it changes some codes in such hidden interned value, I was intending to add a regression test for it. Other case are pre-existing